File: pages/pages/open_account_page.rb
# frozen_string_literal: true

module Pages
  class OpenAccountPage < SitePrism::Page
    element :account_type_dropdown, '#type'
    element :from_account_dropdown, '#fromAccountId'
    element :open_new_account_button, :xpath, '//input[@value="Open New Account"]'
    element :new_account_id_text, '#newAccountId'

    def click_open_account_button
      open_new_account_button.click
    end

    def return_new_account_id
      new_account_id_text.text
    end

    def open_account(type)
      account_type_dropdown.click
      find(:xpath, "//option[text()='#{type}']")
      click_open_account_button
      return_new_account_id
    end
  end
end

File: pages/open_account_page.py
class OpenAccountPage:
    def __init__(self, page):
        self.account_type_dropdown = page.locator("#type")
        self.from_account_dropdown = page.locator("#fromAccountId")
        self.open_new_account_button = page.locator("//input[@value='Open New Account']")
        self.new_account_id_text = page.locator("#newAccountId")

    def click_open_account_button(self):
        self.open_new_account_button.click()

    def return_new_account_id(self):
        return self.new_account_id_text.text_content()

    def open_account(self, type):
        self.account_type_dropdown.select_option(type)
        self.click_open_account_button()
        return self.return_new_account_id()
